Objects/typeobject.c

```diff
@@ -7782,61 +7782,38 @@ slot_tp_getattro(PyObject *self, PyObject *name)
     return vectorcall_method(&_Py_ID(__getattribute__), stack, 2);
 }
 
-static PyObject *
-call_attribute(PyObject *self, PyObject *attr, PyObject *name)
-{
-    PyObject *res, *descr = NULL;
-    descrgetfunc f = Py_TYPE(attr)->tp_descr_get;
-
-    if (f != NULL) {
-        descr = f(attr, self, (PyObject *)(Py_TYPE(self)));
-        if (descr == NULL)
-            return NULL;
-        else
-            attr = descr;
-    }
-    res = PyObject_CallOneArg(attr, name);
-    Py_XDECREF(descr);
-    return res;
-}
-
 static PyObject *
 slot_tp_getattr_hook(PyObject *self, PyObject *name)
 {
     PyTypeObject *tp = Py_TYPE(self);
     PyObject *getattr, *getattribute, *res;
+    PyObject *args[] = { self, name };
+    PyThreadState *tstate = _PyThreadState_GET();
 
-    /* speed hack: we could use lookup_maybe, but that would resolve the
-       method fully for each attribute lookup for classes with
-       __getattr__, even when the attribute is present. So we use
-       _PyType_Lookup and create the method only when needed, with
-       call_attribute. */
-    getattr = _PyType_Lookup(tp, &_Py_ID(__getattr__));
```

Comment:
I just realised what these speed hacks are talking about. For the path where one of the methods isn't required, this is significantly faster as it doesn't bother with the descriptor protocol for those unused methods.

We should re-add the speed hacks.